### DISCUSSION TITLE
how to run the code & environment?

### DISCUSSION THREAD

**Discussion by bodyduardU** [community/community#42237] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello everyone!
I study ArcGis for map Edit now am I am a begginer for ArcGis development but not for programming.
It's difficult to run the souce-the course is simple code.
So i don't know how to run code.
I want your help!!!Thank Q

**Comment by getlost01** [comment]
To run code in ArcGis, you will need to use the Python window in ArcMap or ArcGIS Pro. Here's how to do it:

1. Open ArcMap or ArcGIS Pro and start a new project.
2. Click on the "Geoprocessing" tab in the ArcMap or ArcGIS Pro ribbon.
3. In the "Geoprocessing" tab, click on the "Python" button to open the Python window.
4. In the Python window, you can enter your Python code and press "Enter" to run it.

For example, if you have a script saved as "C:\Scripts\MyScript.py", you can run it in the Python window by typing the following command and pressing "Enter":
```bash
execfile(r"C:\Scripts\MyScript.py")
```
Alternatively, you can also run your script from the command line by using the Python interpreter. To do this, open a command prompt and navigate to the directory where your script is saved. Then, type the following command and press "Enter":
```bash
python MyScript.py
```
I hope this helps!

**Reply by bodyduardU** [reply]
Thank you for your help.I have one more question-the arcgis runs only with python?
Tnak you.
